src/components/Inputs/BaseInput.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import {
+  View,
+  TextInput,
+  StyleSheet,
+  Image,
+  TextInputProps,
+} from 'react-native';
+
+import EyeIcon from '../../icons/Eye';
+import EyeClosedIcon from '../../icons/EyeClosed';
+import { Search as SearchIcon } from '../../icons/Search';
+import ClearInputIcon from '../../icons/ClearInput';
+
+
+type Props = {
+  placeholder: string;
+  children?: React.ReactNode;
+  type?: 'email' | 'phone';
+  iconPosition?: 'left' | 'right';
+  icon?: React.ReactNode
+} & TextInputProps;
+
+const BaseInput = ({
+  placeholder,
+  children,
+  type = 'email',
+  iconPosition = 'left',
+  icon,
+  ...props
+}: Props) => {
+  const [inputFocused, setInputFocused] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+
+  const keyboardType =
+    type === 'phone' ? 'phone-pad' : type === 'email' ? 'email-address' : 'default';
+  const iconSide = iconPosition === 'left' ? styles.iconLeft : styles.iconRight;
+
+  const iconMap = {
+    Eye: EyeIcon,
+    EyeClosed: EyeClosedIcon,
+    Search: SearchIcon,
+    ClearInput: ClearInputIcon,
+  };
+
+  return (
+    <View style={[styles.inputContainer, inputFocused && styles.focusedInput]}>
+      {iconPosition === 'left' && icon && (
+       icon
+      )}
+      
+      <TextInput
+        {...props}
+        style={styles.input}
+        placeholder={placeholder}
+        placeholderTextColor="gray"
+        selectionColor="#B34AFF"
+        value={inputValue}
+        onChangeText={setInputValue}
+        onFocus={() => setInputFocused(true)}
+        onBlur={() => setInputFocused(false)}
+        keyboardType={keyboardType}
+      />
+      {iconPosition === 'right' && icon && (
+       icon
+      )}
+      {children}
```

Comment:
What is the point of adding children here? Do you expect that we pass elements to this input component to be rendered alongside it?

Reply:
Done